### PR TITLE
Fix an issue kube local proxy requirement is wrong in separate port mode

### LINKED ISSUE #27618: Regression with `tsh kube login` in tsh 13.0.4?
When running `tsh kube login $cluster` in 13.0.3 this is the output

```
Logged into Kubernetes cluster "redacted". Try 'kubectl version' to test the connection.
```
which also has Teleport update the kubeconfig file with the appropiate config to be able to run `kubectl` commands against a cluster "out-of-the-box".

**Current behavior on tsh >= 13.0.4:**

Running `tsh kube login $cluster` in 13.0.4 or 13.1.0 outputs the following, requiring users to spin up a local proxy in order to connect to a Kubernetes cluster. Alternatively users can also run commands via `tsh kubectl`.

```
Logged into Kubernetes cluster "redacted". Start the local proxy:
  tsh proxy kube -p 8443
```

As far as I understand, this behaviour was introduced on tsh 13.0.4 via https://github.com/gravitational/teleport/pull/26305 to support running Teleport behind an ALB, but our Teleport cluster is running behind a NLB and there were no issues using `tsh kube login` and then running `kubectl` commands without the need to spin up a local proxy or use `tsh kubectl` prior to tsh 13.0.4.

The question would be if this is an expected behavior from here on out or is it a regression introduced on 13.0.4? Since this looks like a breaking change for end users.

Cheers!

Bug details:
- Teleport cluster version 13.1.0
- tsh versions 13.0.3 VS >=13.0.4
- Setup details:
    - Teleport deployed on EKS with Helm chart on aws mode and running behind a NLB 

<details>
<summary>Logs for <b>tsh login --debug</b> on tsh 13.1.0</summary>

```2023-06-08T13:31:07+02:00 DEBU [KEYSTORE]  Reading certificates from path "/Users/redacted/.tsh/keys/redacted/redacted-ssh/redacted-cert.pub". client/keystore.go:341
2023-06-08T13:31:07+02:00 DEBU [KEYSTORE]  Reading certificates from path "/Users/redacted/.tsh/keys/redacted/redacted-kube/redacted". client/keystore.go:341
2023-06-08T13:31:07+02:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-06-08 19:03:20 +0000 UTC". client/client_store.go:91
2023-06-08T13:31:07+02:00 INFO [CLIENT]    ALPN connection upgrade required for "tredacted:443": true. client/api.go:643
2023-06-08T13:31:07+02:00 INFO [CLIENT]    no host login given. defaulting to redacted client/api.go:1006
2023-06-08T13:31:07+02:00 INFO [CLIENT]    [KEY AGENT] Connected to the system agent: "/private/tmp/com.apple.launchd.f8En7rRjvM/Listeners" client/api.go:4256
2023-06-08T13:31:07+02:00 DEBU [KEYSTORE]  Reading certificates from path "/Users/redacted/.tsh/keys/redacted/redacted-ssh/redacted-cert.pub". client/keystore.go:341
2023-06-08T13:31:07+02:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-06-08 19:03:20 +0000 UTC". client/client_store.go:91
2023-06-08T13:31:07+02:00 INFO [KEYAGENT]  Loading SSH key for user "redacted" and cluster "redacted". client/keyagent.go:195
2023-06-08T13:31:07+02:00 DEBU [CLIENT]    not using loopback pool for remote proxy addr: redacted:443 client/api.go:4211
2023-06-08T13:31:07+02:00 DEBU             Attempting GET redacted:443/webapi/ping/github webclient/webclient.go:128
> Profile URL:        https://redacted:443
  Logged in as:       redacted
  Cluster:            redacted
  Roles:              redacted
  Traits:             github_teams: [redacted]
                      kubernetes_groups: []
                      kubernetes_users: []
                      logins: [redacted]
  Logins:             redacted
  Kubernetes:         enabled
  Kubernetes cluster: "redacted"
  Kubernetes groups:  system:masters
  Valid until:        2023-06-08 21:03:20 +0200 CEST [valid for 7h32m0s]
  Extensions:         login-ip, permit-agent-forwarding, permit-port-forwarding, permit-pty, private-key-policy
```
</details>

<details>
<summary>Logs for <b>tsh kube login $cluster --debug</b> on tsh 13.1.0</summary>

```2023-06-08T13:04:07+02:00 INFO [CLIENT]    ALPN connection upgrade required for "redacted:443": true. client/api.go:643
2023-06-08T13:04:07+02:00 INFO [CLIENT]    no host login given. defaulting to redacted client/api.go:1006
2023-06-08T13:04:07+02:00 INFO [CLIENT]    [KEY AGENT] Connected to the system agent: "/private/tmp/com.apple.launchd.f8En7rRjvM/Listeners" client/api.go:4256
2023-06-08T13:04:07+02:00 DEBU [KEYSTORE]  Reading certificates from path "/Users/redacted/.tsh/keys/redacted/redacted-ssh/redacted-cert.pub". client/keystore.go:341
2023-06-08T13:04:07+02:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-06-08 19:03:20 +0000 UTC". client/client_store.go:91
2023-06-08T13:04:07+02:00 INFO [KEYAGENT]  Loading SSH key for user "redacted" and cluster "redacted". client/keyagent.go:195
2023-06-08T13:04:07+02:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-06-08 19:03:20 +0000 UTC". client/client_store.go:91
2023-06-08T13:04:07+02:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-06-08 19:03:20 +0000 UTC". client/client_store.go:91
2023-06-08T13:04:07+02:00 INFO [CLIENT]    Connecting to proxy=redacted:3023 login="redacted" client/api.go:3040
2023-06-08T13:04:07+02:00 DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:195
2023-06-08T13:04:08+02:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-06-08 19:03:20 +0000 UTC". client/client_store.go:91
2023-06-08T13:04:08+02:00 DEBU [KEYAGENT]  "Checking key: ssh-rsa-cert-v01@openssh.com redacted\n." client/keyagent.go:367
2023-06-08T13:04:08+02:00 DEBU [KEYAGENT]  Validated host redacted:3023. client/keyagent.go:373
2023-06-08T13:04:08+02:00 INFO [CLIENT]    Successful auth with proxy redacted:3023. client/api.go:3048
2023-06-08T13:04:08+02:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-06-08 19:03:20 +0000 UTC". client/client_store.go:91
2023-06-08T13:04:08+02:00 DEBU [CLIENT]    Client  is connecting to auth server on cluster "redacted". client/client.go:1281
2023-06-08T13:04:08+02:00 DEBU [CLIENT]    not using loopback pool for remote proxy addr: redacted:443 client/api.go:4211
2023-06-08T13:04:08+02:00 DEBU             Attempting GET redacted:443/webapi/ping/github webclient/webclient.go:128
2023-06-08T13:04:09+02:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-06-08 19:03:20 +0000 UTC". client/client_store.go:91
2023-06-08T13:04:09+02:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-06-08 19:03:20 +0000 UTC". client/client_store.go:91
2023-06-08T13:04:09+02:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-06-08 19:03:20 +0000 UTC". client/client_store.go:91
2023-06-08T13:04:09+02:00 INFO [CLIENT]    Connecting to proxy=redacted:3023 login="redacted" client/api.go:3040
2023-06-08T13:04:09+02:00 DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:195
2023-06-08T13:04:09+02:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-06-08 19:03:20 +0000 UTC". client/client_store.go:91
2023-06-08T13:04:09+02:00 DEBU [KEYAGENT]  "Checking key: ssh-rsa-cert-v01@redacted\n." client/keyagent.go:367
2023-06-08T13:04:09+02:00 DEBU [KEYAGENT]  Validated host redacted:3023. client/keyagent.go:373
2023-06-08T13:04:09+02:00 INFO [CLIENT]    Successful auth with proxy redacted:3023. client/api.go:3048
2023-06-08T13:04:09+02:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-06-08 19:03:20 +0000 UTC". client/client_store.go:91
2023-06-08T13:04:09+02:00 DEBU [CLIENT]    Client  is connecting to auth server on cluster "redacted". client/client.go:1281
2023-06-08T13:04:09+02:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-06-08 19:03:20 +0000 UTC". client/client_store.go:91
2023-06-08T13:04:09+02:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-06-08 19:03:20 +0000 UTC". client/client_store.go:91
2023-06-08T13:04:09+02:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-06-08 19:03:20 +0000 UTC". client/client_store.go:91
2023-06-08T13:04:09+02:00 INFO [CLIENT]    Connecting to proxy=redacted:3023 login="redacted" client/api.go:3040
2023-06-08T13:04:09+02:00 DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:195
2023-06-08T13:04:09+02:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-06-08 19:03:20 +0000 UTC". client/client_store.go:91
2023-06-08T13:04:09+02:00 DEBU [KEYAGENT]  "Checking key: ssh-rsa-cert-v01@openssh.com redacted\n." client/keyagent.go:367
2023-06-08T13:04:09+02:00 DEBU [KEYAGENT]  Validated host redacted:3023. client/keyagent.go:373
2023-06-08T13:04:10+02:00 INFO [CLIENT]    Successful auth with proxy redacted:3023. client/api.go:3048
2023-06-08T13:04:10+02:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-06-08 19:03:20 +0000 UTC". client/client_store.go:91
2023-06-08T13:04:10+02:00 DEBU [CLIENT]    Client  is connecting to auth server on cluster "redacted". client/client.go:1281
```
</details>

### PR DESCRIPTION
**Pull request by greedy52**
Fixes #27618 
- #27618 

Introduced in #26305

### CODE REVIEW

File: tool/tsh/common/kube.go [hunk 1]
```diff
@@ -773,8 +773,8 @@ func isNetworkError(err error) bool {
 	return errors.As(err, &opErr) || trace.IsConnectionProblem(err)
 }
 
-func (c *kubeCredentialsCommand) checkLocalProxyRequirement(connUpgradeRequired bool) error {
-	if connUpgradeRequired {
+func (c *kubeCredentialsCommand) checkLocalProxyRequirement(profile *profile.Profile) error {
```
Comment (greedy52) [inline tool/tsh/common/kube.go:776]: fix for `tsh kube credentials`

File: tool/tsh/common/kube.go [hunk 2]
```diff
@@ -1183,7 +1183,7 @@ func (c *kubeLoginCommand) run(cf *CLIConf) error {
 }
 
 func (c *kubeLoginCommand) printUserMessage(cf *CLIConf, tc *client.TeleportClient) {
-	if c.localProxyRequired(tc) {
+	if profileRequireKubeLocalProxy(tc.Profile()) {
```
Comment (greedy52) [inline tool/tsh/common/kube.go:1186]: fix for `tsh kube login`

File: tool/tsh/common/kubectl.go [hunk 3]
```diff
@@ -470,9 +476,7 @@ func shouldUseKubeLocalProxy(cf *CLIConf, kubectlArgs []string) (*clientcmdapi.C
 		return nil, nil, false
 	}
 
-	// Only use local proxy when Kube is enabled and ALPN connection upgrade is
-	// required.
-	if !profile.TLSRoutingConnUpgradeRequired || profile.KubeProxyAddr == "" {
+	if !profileRequireKubeLocalProxy(profile) {
```
Comment (greedy52) [inline tool/tsh/common/kubectl.go:479]: fix for `tsh kubectl` and `tsh kube exec`

**Review by smallinsky — APPROVED** [review]

File: tool/tsh/common/kubectl.go [hunk 4]
```diff
@@ -452,12 +452,18 @@ func makeAndStartKubeLocalProxy(cf *CLIConf, config *clientcmdapi.Config, cluste
 	return closeFn, localProxy.KubeConfigPath(), nil
 }
 
+// profileRequireKubeLocalProxy returns true if provided profile indicates a
+// local proxy is required for kube access.
+func profileRequireKubeLocalProxy(profile *profile.Profile) bool {
+	return profile.KubeProxyAddr == profile.WebProxyAddr && profile.TLSRoutingConnUpgradeRequired
+}
```
Comment (tigrato) [inline tool/tsh/common/kubectl.go:459]: Can we refactor this into profile method?

`profile.RequiresKubeLocalProxy()` ?

**Review by tigrato — APPROVED** [review]

**Pull request merged**